Assignment 4

jwoolbright23 opened the pull request.

jwoolbright23 committed:
added directories for project structure

jwoolbright23 committed:
completed initial structure for project directory layout

jwoolbright23 committed:
added intro and starter code sections, will need to update links when able

jwoolbright23 committed:
updates to intro through part three complete

jwoolbright23 committed:
migration of part-4 complete

updates to starter-code and part-two links

jwoolbright23 committed:
added submission instruction section

jwoolbright23 committed:
fixed headers on submission page

jwoolbright23 committed:
rebase

codinglikeagirl42 commented on content/assignments/techjobs-persistent/_index.md:
```diff
@@ -0,0 +1,28 @@
+date: 2023-01-03T10:42:56-06:00
+draft: false
+weight: 2
+originalAuthor: Sally Steuterman # to be set by page creator
+originalAuthorGitHub: gildedgardenia # to be set by page creator
+reviewer: Kimberly Horan # to be set by the page reviewer
+reviewerGitHub: codinglikeagirl42 # to be set by the page reviewer
```
```suggestion
originalAuthor: John Woolbright # to be set by page creator
originalAuthorGitHub: jwoolbright23 # to be set by page creator
reviewer: Kimberly Horan # to be set by the page reviewer
reviewerGitHub: codinglikeagirl42 # to be set by the page reviewer
```

codinglikeagirl42 commented on content/assignments/techjobs-persistent/part-four/index.md:
```diff
@@ -0,0 +1,81 @@
+## Updating `HomeController`, Again
+
+You next need to wire `HomeController` with the skills data in order to add skills objects to a new job.
+This will look almost precisely like what you have done for employer data above. Refer back to [that section](http://education.launchcode.org/java-web-dev-curriculum/assignments/techjobs-persistent/part-three/#updating-homecontroller) to inject the controller with skill data.
```
Add a todo to fix link after merge.

codinglikeagirl42 commented on content/assignments/techjobs-persistent/starter-code-review/index.md:
```diff
@@ -0,0 +1,50 @@
+
+Set up a local copy of the project:
+
+1. In Canvas, **Graded Assignment #4: TechJobs (Persistent Edition)** contains a GitHub Classroom assignment invitation link and then set up the project in IntelliJ. Refer back to the GitHub Classroom instructions from [assignment0](https://education.launchcode.org/java-web-dev-curriculun/assignments/hello-world/index.html) for details. 
```
Update wording to remove Github Classroom reference.

Add todo to update link

codinglikeagirl42 replied:
```suggestion
<!-- TODO: Link below will need to be updated once merged -->
1. In Canvas, **Graded Assignment #4: TechJobs (Persistent Edition)** contains a GitHub Classroom assignment invitation link and then set up the project in IntelliJ. Refer back to the GitHub Classroom instructions from [assignment0]() for details. 
```

codinglikeagirl42 commented on content/assignments/techjobs-persistent/starter-code-review/index.md:
```diff
@@ -0,0 +1,50 @@
+1. The tests for this assignment are set up the same way as for [assignment 2](https://education.launchcode.org/java-web-development/assignments/tech-jobs-oo.html#assignment-2-autograding).
+
+You won't be able to run your application or the tests yet. If you try to do so, you'll see a host of errors relating to the
+Spring Data annotations and classes. Some of these have already been used in the code, but the dependency that includes them has not yet been declared. That will be one of your tasks. You'll need to complete [Part1](https://education.launchcode.org/java-web-dev-curriculum/assignments/techjobs-persistent/part-one/#connect-a-database-to-a-spring-app) before you can
```
```suggestion
Spring Data annotations and classes. Some of these have already been used in the code, but the dependency that includes them has not yet been declared. That will be one of your tasks. You'll need to complete [Part1]({{< relref "../part-one" >}}) before you can
```

codinglikeagirl42 commented on content/assignments/techjobs-persistent/starter-code-review/index.md:
```diff
@@ -0,0 +1,50 @@
+
+That said, it's a good idea to scan the classes and templates even before you're able to execute
+`bootRun`. Take a gander at the `Job` class. It will look somewhat similar to the model in
+[Tech Jobs (MVC Edition)](https://education.launchcode.org/java-web-development/assignments/tech-jobs-mvc.html#tech-jobs-mvc), with a few key differences.
```
Add todo to update link to new hugo book after merge

codinglikeagirl42 replied:
```suggestion
<!-- TODO: Link below will need to be updated once assignment 3 is in the book -->
[Tech Jobs (MVC Edition)](), with a few key differences.
```

codinglikeagirl42 commented on content/assignments/techjobs-persistent/starter-code-review/index.md:
```diff
@@ -0,0 +1,50 @@
+1. In Canvas, **Graded Assignment #4: TechJobs (Persistent Edition)** contains a GitHub Classroom assignment invitation link and then set up the project in IntelliJ. Refer back to the GitHub Classroom instructions from [assignment0](https://education.launchcode.org/java-web-dev-curriculun/assignments/hello-world/index.html) for details. 
+1. Launch the application (via the Gradle pane, *Tasks > Application > bootRun*) to make sure it starts up properly. Then shut it down.
+<!-- TODO: Link below will need to be updated once assignment 2 is in the book -->
+1. The tests for this assignment are set up the same way as for [assignment 2](https://education.launchcode.org/java-web-development/assignments/tech-jobs-oo.html#assignment-2-autograding).
```
```suggestion
1. The tests for this assignment are set up the same way as for [assignment 2]().
```

codinglikeagirl42 commented on content/assignments/techjobs-persistent/starter-code-review/index.md:
```diff
@@ -0,0 +1,50 @@
+[Tech Jobs (MVC Edition)](https://education.launchcode.org/java-web-development/assignments/tech-jobs-mvc.html#tech-jobs-mvc), with a few key differences.
+
+You're no longer using a csv file to load job data, instead, we'll be creating new Job objects via a
+user form. The Job data will be stored in a MySQL database that you'll setup in [Part 1](http://education.launchcode.org/java-web-dev-curriculum/assignments/techjobs-persistent/part-one/#connect-a-database-to-a-spring-app) of this assignment.
```
```suggestion
user form. The Job data will be stored in a MySQL database that you'll setup in [Part 1]({{< relref "../part-one/#connect-a-database-to-a-spring-app" >}}) of this assignment.
```

codinglikeagirl42 commented on content/assignments/techjobs-persistent/starter-code-review/index.md:
```diff
@@ -0,0 +1,50 @@
+
+As you explore
+the starter code, you'll notice that the `JobField` abstract class is no longer present. Your task for
+[Part 2](https://education.launchcode.org/java-web-dev-curriculum/assignments/techjobs-persistent/part-two/#persisting-employers-and-skills) is to complete the work to persist some of the classes.
```
```suggestion
[Part 2]({{< relref "../part-two/#persisting-employers-and-skills" >}}) is to complete the work to persist some of the classes.
```

codinglikeagirl42 commented on content/assignments/techjobs-persistent/starter-code-review/index.md:
```diff
@@ -0,0 +1,50 @@
+[Part 2](https://education.launchcode.org/java-web-dev-curriculum/assignments/techjobs-persistent/part-two/#persisting-employers-and-skills) is to complete the work to persist some of the classes.
+You'll do this for `Employer` and `Skill` classes, as well as `Job`.
+
+The `Job` class will also look different from how you have last seen it. In Parts 3 and 4, you'll
```
```suggestion
The `Job` class will also look different from how you have last seen it. In [Part 3]({{< relref "../part-three/#setting-up-a-one-to-many-relationship" >}}) and [Part 4]({{< relref "../part-four/#setting-up-a-many-to-many-relationship" >}}), you'll
```

codinglikeagirl42 commented on content/assignments/techjobs-persistent/submission/index.md:
```diff
@@ -0,0 +1,23 @@
+
+## How to Submit
+
+To turn in your assignment and get credit, follow the [Submission Instructions](https://education.launchcode.org/java-web-dev-curriculun/assignments/hello-world/index.html#submitting-your-work-on-canvas)
```
Add todo to update after merging.

codinglikeagirl42 replied:
```suggestion
<!-- TODO: Link below will need to be updated once merged -->

To turn in your assignment and get credit, follow the [Submission Instructions]()
```

codinglikeagirl42 commented on content/assignments/techjobs-persistent/submission/index.md:
```diff
@@ -0,0 +1,23 @@
+
+## How to Submit
+
+To turn in your assignment and get credit, follow the [Submission Instructions](https://education.launchcode.org/java-web-dev-curriculun/assignments/hello-world/index.html#submitting-your-work-on-canvas)
```
```suggestion
To turn in your assignment and get credit, follow the [Submission Instructions]()
```

codinglikeagirl42 commented on content/assignments/techjobs-persistent/part-one/index.md:
```diff
@@ -0,0 +1,52 @@
+`spring.datasource.url` set to the address of your database connection, as well as the username and password for a user you have create Refer to the tip below for the other properties you must add to complete your database setup.
+
+{{% notice blue Tip "rocket" %}}
+You can double check your setup against what you've already done for [your coding events repo](https://education.launchcode.org/java-web-development/chapters/orm-part1/background.html#setup-orm-database). You can copy these property assignments from your coding events repo, only needing to change the database address and username/password values.
```
```suggestion
<!-- TODO: Link below will need to be updated once chapter 17 is merged in -->

You can double check your setup against what you've already done for [your coding events repo](). You can copy these property assignments from your coding events repo, only needing to change the database address and username/password values.
```

codinglikeagirl42 replied:
Add todo to update when chapter 17 merged into the book.

codinglikeagirl42 commented on content/assignments/techjobs-persistent/part-two/index.md:
```diff
@@ -0,0 +1,105 @@
+
+## Persisting Employers and Skills
+
+You will need to have completed the [setup steps](https://education.launchcode.org/java-web-dev-curriculum/assignments/techjobs-persistent/part-one/#connect-a-database-to-a-spring-app) before starting this
```
```suggestion
You will need to have completed the [setup steps]({{< relref "../part-one/#connect-a-database-to-a-spring-app" >}}) before starting this
```

codinglikeagirl42 commented on content/assignments/techjobs-persistent/part-two/index.md:
```diff
@@ -0,0 +1,105 @@
+
+1. Open the `Employer` model class. In addition to the fields inherited from `AbstractEntity`, `Employer` should have a string field for `location`. Add the field for `location` with validation that ensures it is not empty and has a reasonable length. In addition, add public accessor methods to `Employer`.
+
+{{% notice blue Note "rocket" %}}
+For the purposes of this application, an employer can only have one location.
+{{% /notice %}}
```
```suggestion
   {{% notice blue Note "rocket" %}}
   For the purposes of this application, an employer can only have one location.
   {{% /notice %}}
```

jwoolbright23 replied:
This applies a strange font / box when tabbed in. Going to leave as is for now unless there is a fix I am unaware of.

jwoolbright23 replied:
Ordered the numbers manually as a fix.

codinglikeagirl42 commented on content/assignments/techjobs-persistent/part-one/index.md:
```diff
@@ -0,0 +1,52 @@
+
+1. Start MySQL Workbench and create a new schema named `techjobs`.
+
+{{% notice blue Tip "rocket" %}}
+Remember to double click on the schema name in the file tree to make it the default schema.
+{{% /notice %}}
```
```suggestion
   {{% notice blue Tip "rocket" %}}
   Remember to double click on the schema name in the file tree to make it the default schema.
   {{% /notice %}}
```

codinglikeagirl42 replied:
Formatting, tab in tip so ordered list formats correctly 1,2,3,4

jwoolbright23 replied:
Also ordered the numbers manually here to avoid funky formatting within note

codinglikeagirl42 commented on content/assignments/techjobs-persistent/part-three/index.md:
```diff
@@ -0,0 +1,75 @@
+The *List* and *Search* functionality still isn't quite fixed so to view a job in the application, make a note
+of the job's id in the SQL table. Back in your browser, enter the path for `/view/{jobId}`.
+
+When everything works, move on to Part 4 below.
```
```suggestion
When everything works, move on to [ Part 4]({{< relref "../part-four" >}}).
```

codinglikeagirl42 commented on content/assignments/techjobs-persistent/part-one/index.md:
```diff
@@ -0,0 +1,52 @@
+
+Your running application still has limited functionality. You won't yet be able to add a job with the *Add Job* form. You also
+won't yet be able to view the list of jobs or search for jobs - but this is mostly because you have no job data. Move on to
+Part 2 below to start adding these functionalities.
```
```suggestion
[Part 2]({{< relref "../part-two" >}}) to start adding these functionalities.
```

codinglikeagirl42 reviewed (approved):
Review comments - accept suggestions and double check against trello checklist.

jwoolbright23 committed:
updated links

updated all links that were linkable, commented out all links that are currently unavailable
